atc/api/teamserver/set.go

```diff
@@ -43,11 +43,13 @@ func (s *Server) SetTeam(w http.ResponseWriter, r *http.Request) {
 
 	if found {
 		hLog.Debug("updating-credentials")
-		err = team.UpdateProviderAuth(atcTeam.Auth)
-		if err != nil {
-			hLog.Error("failed-to-update-team", err, lager.Data{"teamName": teamName})
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+		if len(atcTeam.Auth) > 0 {
```

Comment:
I suspect we could invalidate the request without even looking up the team, and save an extra level of indentation while we're at it.

I'm just thinking out loud too (and this suggestion might constitute "over-engineering") but I could imagine a `Validate() error` method on the `atc.Team` type, that gets called right after decoding it [up here](https://github.com/concourse/concourse/blob/0b79feeb5ab7fa205c60ae6d34334152390cc74c/atc/api/teamserver/set.go#L24-L29), looking something like

```go
	err = atcTeam.Validate()
	if err != nil {
		hLog.Error("invalid-team", err)
		w.WriteHeader(http.StatusBadRequest)
		return
	}
```

Reply:
@jamieklassen  yes at first look yesterday I did not have better understanding of what the code does. After looking carefully I forsee changes in two places:
1. Fly set-team command is validating values for command line auth params. so it should do the same for config file too see commit [here](https://github.com/volkswagen/concourse/commit/7693ba2daf039ba18b94883521e0a16dd44b6d47#diff-d5c54d5d65e15d27255ae063f0a616a5)

2. In order to be defensive its also good idea to validate in the backend, to be sure that auth config is never empty. See commit [here](https://github.com/volkswagen/concourse/commit/32014c630ad4124e521130daa79487fc19bb49f1)

I really like the validate method, its more readable than check if length is zero. If the CI passes i will do this refactor as well :)

Reply:
nice catch on the server side! Smart to do the same validation for `--main-team-config`, as discussed in concourse/docs#319.